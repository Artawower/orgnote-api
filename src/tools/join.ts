export function join(...path: string[]): string {
  return path.join('/').replace(/\/+/g, '/').replace(/\/+$/, '');
}
