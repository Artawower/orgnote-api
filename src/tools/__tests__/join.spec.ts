import { test, expect } from 'vitest';
import { join } from '../join';

test('Should join file paths', () => {
  const samples = [
    ['dir1', 'subdir2', 'file'],
    ['dir2/', '/file'],
    ['/dri3', 'subdir2/', 'file'],
    ['file'],
  ];

  samples.forEach((sample) => {
    const result = join(...sample);
    expect(result).toMatchSnapshot();
  });
});
