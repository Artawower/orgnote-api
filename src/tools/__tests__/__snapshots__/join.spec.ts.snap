// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should join file paths 1`] = `"dir1/subdir2/file"`;

exports[`Should join file paths 2`] = `"dir2/file"`;

exports[`Should join file paths 3`] = `"/dri3/subdir2/file"`;

exports[`Should join file paths 4`] = `"file"`;
